commands/database-update.js

```diff
@@ -35,31 +34,34 @@ module.exports = new Command('database:update <path> [infile]')
         return utils.reject('Command aborted.', {exit: 1});
       }
 
-      return new RSVP.Promise(function(resolve, reject) {
-        var inStream = utils.stringToStream(options.data) || (infile ? fs.createReadStream(infile) : process.stdin);
+      var inStream = utils.stringToStream(options.data) || (infile ? fs.createReadStream(infile) : process.stdin);
+      var url = utils.addSubdomain(api.realtimeOrigin, options.instance) + path + '.json?';
 
-        var url = utils.addSubdomain(api.realtimeOrigin, options.instance) + path + '.json?';
-        var query = {auth: options.databaseAdminToken};
-
-        url += querystring.stringify(query);
+      if (!infile && !options.data) {
+        utils.explainStdin();
+      }
 
-        if (!infile && !options.data) {
-          utils.explainStdin();
-        }
+      var reqOptions = {
+        url: url,
+        json: true
+      };
 
-        inStream.pipe(request.patch(url, {json: true}, function(err, res, body) {
-          logger.info();
-          if (err) {
-            return reject(new FirebaseError('Unexpected error while setting data', {exit: 2}));
-          } else if (res.statusCode >= 400) {
-            return reject(responseToError(res, body));
-          }
+      return api.addAccessTokenToHeader(reqOptions).then(function() {
```

Comment:
Same again, I think you should use the return value from the promise rather than relying on the original object being mutated - it makes it easier to read